src/aiida/tools/dumping/processes.py

```diff
@@ -0,0 +1,454 @@
+        )
+
+        # `verdi process status`
+        process_status = format_call_graph(calc_node=process_node, max_depth=None, call_link_label=True)
+        _readme_string += f'\n\n\nOutput of `verdi process status {process_node.pk}:`\n\n{process_status}'
+
+        # `verdi process report`
+        # Copied over from `cmd_process`
+        if isinstance(process_node, CalcJobNode):
+            process_report = get_calcjob_report(process_node)
+        elif isinstance(process_node, WorkChainNode):
+            process_report = get_workchain_report(process_node, levelname='REPORT', indent_size=2, max_depth=None)
+        elif isinstance(process_node, (CalcFunctionNode, WorkFunctionNode)):
+            process_report = get_process_function_report(process_node)
+        else:
+            process_report = f'Nothing to show for node type {process_node.__class__}'
+
+        _readme_string += f'\n\n\nOutput of `verdi process report {process_node.pk}`:\n\n{process_report}'
+
+        # `verdi process show`?
+        process_show = get_node_info(node=process_node)
+        _readme_string += f'\n\n\nOutput of `verdi process show {process_node.pk}`:\n\n{process_show}'
+
+        with (output_path / 'README').open('w') as handle:
+            handle.write(_readme_string)
+
+    @staticmethod
+    def _generate_child_node_label(index: int, link_triple: LinkTriple) -> str:
+        """Small helper function to generate and clean directory label for child nodes during recursion.
+
+        :param index: Index assigned to step at current level of recursion.
+        :param link_triple: `LinkTriple` of `ProcessNode` explored during recursion.
+        :return: Chlild node label during recursion.
+        """
+        node = link_triple.node
+        link_label = link_triple.link_label
+
+        # Generate directories with naming scheme akin to `verdi process status`
+        label_list = [f'{index:02d}', link_label]
+
+        try:
+            process_label = node.process_label
+            if process_label is not None and process_label != link_label:
+                label_list += [process_label]
+
+        except AttributeError:
+            process_type = node.process_type
+            if process_type is not None and process_type != link_label:
+                label_list += [process_type]
+
+        node_label = '-'.join(label_list)
+        # `CALL-` as part of the link labels also for MultiplyAddWorkChain -> Seems general enough, so remove
+        node_label = node_label.replace('CALL-', '')
+        node_label = node_label.replace('None-', '')
+
+        return node_label
+
+    def dump(
+        self,
+        process_node: ProcessNode,
+        output_path: Path | None,
+        io_dump_paths: list | None = None,
+    ) -> Path:
+        """Dumps all data involved in a `ProcessNode`, including its outgoing links.
+
+        Note that if an outgoing link is a `WorkflowNode`, the function recursively calls itself, while files are
```

Comment:
While I appreciate the concept, it does make writing the docstrings even more laborious and I wonder how useful it would be if we don't apply this consistently across all docstrings. At this point, terms like `WorkflowNode` and `ProcessNode` have become part of AiiDA's jargon and the user wouldn't necessarily need the direct link to the API to understand it or be able to work with it.

So at least for me, it would be ok not to necessarily add these explicit links everywhere.

Reply:
Yes, as I mentioned to you @agoscinski, I also think it's a nice concept, thanks for bringing it up. For now, I'd agree with @sphuber, though, and would skip adding them if they are not consistently being used throughout the codebase.